Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskData.kt

```diff
@@ -0,0 +1,22 @@
+package de.rki.coronawarnapp.task
+
+import org.joda.time.Instant
+import java.util.UUID
+
+interface TaskData {
+    val id: UUID
+    val request: TaskRequest<*>
+    val createdAt: Instant
+    val startedAt: Instant?
+    val completedAt: Instant?
+    val error: Throwable?
+    val result: Task.Result?
+    val state: State
+    val type: TaskType
+
+    enum class State {
```

Comment:
How would you distinguish a successfully finished task from a failed finished task?

Reply:
A finished task always has a result, a task that failed, or was canceled will have an exception.